src/main/java/it/pagopa/pm/gateway/controller/PaymentTransactionsController.java

```diff
@@ -205,4 +249,169 @@ private BPayPaymentResponseEntity convertBpayPaymentResponseToEntity(Inserimento
+
+        try {
+            executePostePayPayment(postePayAuthRequest, clientId, paymentRequestEntity);
+        } catch (Exception e){
+            PostePayAuthResponse response =  new  PostePayAuthResponse();
+            response.setChannel(response.getChannel());
+            response.setError("Error during payment authorization request to Postepay");
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(response);
+        }
+
+        log.info("END requestPaymentPostepay " + idTransaction);
+
+        PostePayAuthResponse response = new  PostePayAuthResponse();
+        response.setChannel(response.getChannel());
+        response.setUrlRedirect(PAYMENT_RESPONSE_URLREDIRECT);
+
+        return ResponseEntity.status(HttpStatus.OK).body(response);
+
+    }
+
+    @GetMapping(REQUEST_PAYMENT_POSTEPAY_REQUEST_ID)
+    @ResponseBody
+    public PostePayPollingResponse getPaymentPostepayResponse(@PathVariable String requestId, @RequestHeader(required = false, value = MDC_FIELDS) String mdcFields) throws RestApiException {
+        setMdcFields(mdcFields);
+        PaymentRequestEntity request = paymentRequestRepository.findByGuid(requestId);
+        if (request == null || !REQUEST_PAYMENT_POSTEPAY.equals(request.getRequestEndpoint())) {
+            throw new RestApiException(ExceptionsEnum.TRANSACTION_NOT_FOUND);
+        }
+        return new PostePayPollingResponse(
+                request.getClientId(),
+                request.getAuthorizationUrl(),
+                Boolean.TRUE.equals(request.getAuthorizationOutcome()) ? OK : KO,
+                request.getErrorCode()
+        );
+    }
+
+    @Async
+    private void executePostePayPayment(PostePayAuthRequest postePayAuthRequest, String clientId, PaymentRequestEntity paymentRequestEntity)
+            throws RestApiException, JsonProcessingException {
+        Long idTransaction = postePayAuthRequest.getTransactionId();
+        log.info("START executePostePayPayment for transaction " + idTransaction);
+
+        CreatePaymentRequest createPaymentRequest = mapPostePayAuthRequestToCreatePaymentRequest(postePayAuthRequest, clientId);
+        InlineResponse200 inlineResponse200;
+        try {
+            MicrosoftAzureLoginResponse microsoftAzureLoginResponse = postePayClient.requestMicrosoftAzureLogin();
+            String bearerTokenAuthorization = "Bearer " + microsoftAzureLoginResponse.getAccess_token();
+            inlineResponse200 = paymentManagerControllerApi.apiV1PaymentCreatePost(bearerTokenAuthorization, createPaymentRequest);
+            if (inlineResponse200 == null) {
+                throw new RestApiException(ExceptionsEnum.GENERIC_ERROR);
+            }
+            log.info("Response from PostePay createPayment - idTransaction: " + idTransaction + " - paymentID: "
+                    + inlineResponse200.getPaymentID() + " - userRedirectUrl: " + inlineResponse200.getUserRedirectURL());
+        } catch (ApiException e) {
```

Comment:
There's a fair bit of repetition here, what about something like this?

```suggestion
        } catch (ApiException e) {
            /*
             * Somewhere above
             * boolean exceptionHappened;
             * boolean timeoutExceptionHappened;
             * were defined
             */
             exceptionHappened = true:
        } catch (Exception e) {
            exceptionHappened = true:
            timeoutExceptionHappened = e.getCause() instanceof SocketTimeoutException;
        } finally {
            if (exceptionHappened) {
                log.error("Error from PostePay createPayment: " + error);
                paymentRequestEntity.setAuthorizationOutcome(false);
                
                if (timeoutExceptionHappened) {
                    log.error("SocketTimeoutException during Postepay calling");
                    throw new RestApiException(ExceptionsEnum.TIMEOUT);
                } else {
                    throw new RestApiException(ExceptionsEnum.GENERIC_ERROR);
                }
            }
        }
```

Reply:
This version is more concise than the current one, but i need to save the entity at every step, also i don't like throw inside a finally block. 
By the way your comment led me to clean the code, so added (soon to be pushed ) a separate method for updating and saving the entity to avoid obnoxious code repetition